src/labone/core/value.py

```diff
@@ -287,8 +288,25 @@ def _value_from_python_types(
     Raises:
         LabOneCoreError: If the data type of the value to be set is not supported.
     """
+    value = ann_value.value
     request_value = reflection.Value.new_message()  # type: ignore[attr-defined]
-    if isinstance(value, bool):
+
+    if isinstance(value, np.ndarray):
+        if ann_value.extra_header is None:
+            request_value.vectorData = value.tobytes()
+        else:
+            return {"vectorData": encode_shf_vector_data_struct(
+                value, ann_value.extra_header
+            )}
+    elif isinstance(value, SHFDemodSample):
+        if value.extra_header is None:
+            raise ValueError("SHFDemodSample requires extra_header")
+        return {"vectorData":encode_shf_vector_data_struct(value, ann_value.extra_header)
+        }
+    elif isinstance(value, (TriggerSample, CntSample)):
+        raise NotImplementedError("TriggerSample and CntSample not yet implemented")
+    
+    elif isinstance(value, bool):
```

Comment:
Do you have any idea why changing `return {"vectorData": ...}` to `request_value.vectorData = ...` breaks it?

Reply:
In fact, the logic is needed in this function as well, because it is called from session.set and otherwise the extra_header is lost right away. Is deleting `_value_from_python_types` and always using `_value_from_python_types_dict` a viable solution to you?